safenode/src/storage/registers.rs

```diff
@@ -0,0 +1,735 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::protocol::{
+    messages::{
+        EditRegister, QueryResponse, RegisterCmd, RegisterQuery, ReplicatedRegisterLog,
+        SignedRegisterCreate, SignedRegisterEdit,
+    },
+    types::{
+        address::RegisterAddress,
+        register::{Action, EntryHash, Register, User},
+    },
+};
+
+use super::{
+    register_store::{RegisterStore, StoredRegister},
+    Error, Result,
+};
+
+use bincode::serialize;
+use tracing::{debug, error, info, trace, warn};
+
+/// Operations over the Register data type and its storage.
+#[derive(Clone, Default)]
+pub(super) struct RegisterStorage {
+    register_store: RegisterStore,
```

Comment:
Hmm.... do we need this additional layering? Could we perhaps remove it cleanly? Ideas?